ports/geckolib/gecko_bindings/tools/regen.py

```diff
@@ -0,0 +1,383 @@
+            "nsMargin.h", "nsThemeConstants.h", "nsCSSProperty.h",
+            "CSSVariableValues.h", "nsFont.h", "nsTHashtable.h",
+            "PLDHashTable.h", "nsColor.h", "nsStyleStruct.h", "nsStyleCoord.h",
+            "RefPtr.h", "nsISupportsImpl.h", "gfxFontConstants.h",
+            "gfxFontFamilyList.h", "gfxFontFeatures.h", "imgRequestProxy.h",
+            "nsIRequest.h", "imgIRequest.h", "CounterStyleManager.h",
+            "nsStyleConsts.h", "nsCSSValue.h", "SheetType.h", "nsIPrincipal.h",
+            "nsDataHashtable.h", "nsCSSScanner.h", "Types.h", "utility",
+            "nsTArray", "pair", "SheetParsingMode.h", "StaticPtr.h",
+            "nsProxyRelease.h"
+        ],
+        "blacklist": [
+            "IsDestructibleFallbackImpl", "IsDestructibleFallback",
+            "nsProxyReleaseEvent", "FallibleTArray", "nsTArray_Impl",
+            "__is_tuple_like_impl", "tuple_size", "tuple",
+            "__make_pair_return_impl", "__make_pair_return", "tuple_element",
+            "_Itup_cat"
+        ],
+        "opaque_types": [
+            "nsIntMargin", "nsIntPoint", "nsIntRect", "nsCOMArray",
+            "nsDependentString", "EntryStore", "gfxFontFeatureValueSet",
+            "imgRequestProxy", "imgRequestProxyStatic", "CounterStyleManager",
+            "ImageValue", "URLValue", "URLValueData", "nsIPrincipal",
+            "nsDataHashtable", "imgIRequest"
+        ]
+    },
+    # Generation of the ffi bindings.
+    "bindings": {
+        "raw_lines": [
+            "use heapsize::HeapSizeOf;",
+        ],
+        "match_headers": [
+            "ServoBindings.h",
+            "nsStyleStructList.h",
+        ],
+        "files": [
+            "{}/dist/include/mozilla/ServoBindings.h",
+        ],
+
+        # Types to just use from the `structs` target.
+        "structs_types": [
+            "nsStyleFont", "nsStyleColor", "nsStyleList", "nsStyleText",
+            "nsStyleVisibility", "nsStyleUserInterface", "nsStyleTableBorder",
+            "nsStyleSVG", "nsStyleVariables", "nsStyleBackground",
+            "nsStylePosition", "nsStyleTextReset", "nsStyleDisplay",
+            "nsStyleContent", "nsStyleUIReset", "nsStyleTable",
+            "nsStyleMargin", "nsStylePadding", "nsStyleBorder",
+            "nsStyleOutline", "nsStyleXUL", "nsStyleSVGReset", "nsStyleColumn",
+            "nsStyleEffects", "nsStyleImage", "nsStyleGradient",
+            "nsStyleCoord", "nsStyleGradientStop",
+
+            "SheetParsingMode", "nsMainThreadPtrHandle",
+            "nsMainThreadPtrHolder", "nscolor", "nsFont", "FontFamilyList",
+            "FontFamilyType", "nsIAtom",
+        ],
+    }
+}
+
+
+def platform_dependent_defines():
+    ret = []
+
+    if os.name == "posix":
+        ret.append("-DOS_POSIX=1")
+
+    ret.append({
+        "Linux": "-DOS_LINUX=1",
+        "Darwin": "-DOS_MACOSX=1",
+        # TODO: Windows?
+    }[platform.system()])
+
+    return ret
+
+
+def extend_object(obj, other):
```

Comment:
I don't think so, there's `dict.update(other)`, but that overrides the keys instead of merging them, so we need to write this by hand.